### PR TITLE
Add the Aerosol activation example.

### PR DESCRIPTION
**Pull request by trontrytel**
This adds the Aerosol activation Jupyter notebook example and the corresponding docs.

Co-authored-by: montu12345 <shevali@mercurytrip.com>

### CODE REVIEW

**Comment by trontrytel** [comment]
@odunbar - here is what I have so far regarding the SURF example calibration for aerosol activation. Let me know if that could fit into your docs or if you want me to rewrite something. I'm ok to convert the jupyternotebook to some literate/markdown too, if you decide this doesn't work.

Also, do you guys include the examples in the CI? Would it be possible to include this one too?

**Comment by odunbar** [comment]
@jakebolewski Do you know whether we can a jupyter notebook can be run in CI? Also when does our automated pipeline of examples run in anyway? Cheers. 

I'll look at the PR content later, Thanks!

File: docs/src/examples/Aerosol_activation_example.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+The AerosolActivation model is used here in a "perfect model" setting,
+  meaning that the observations are generated by the same module
+  we are calibrating.
+Also, the setup of the example is very simple
+  and meant to be an entry level tutorial.
+In future aplications the artificially generated observations
+  can be swapped with actual measurements,
+  and the calibration setup should be expanded to include more combinations
+  of atmospheric conditions and initial aerosol size distributions.
+
```
Comment (odunbar) [inline docs/src/examples/Aerosol_activation_example.md:28]: Remove ~~Also, the setup of the example is very simple
  and meant to be an entry level tutorial. In future aplications the artificially generated observations
  can be swapped with actual measurements,
  and the calibration setup should be expanded to include more combinations
  of atmospheric conditions and initial aerosol size distributions.~~

"swapping with actual measurements" requires dealing with model form error, which is likely non-trivial. I would stick to defining what is done here.

File: docs/src/examples/Aerosol_activation_example.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+This example is based on AerosolActivation module which is a part of the
+  [CloudMicrophysics.jl](https://github.com/CliMA/CloudMicrophysics.jl) package.
+The AerosolActivation module computes the total number and mass
+  of aerosol particles that get activated and become cloud droplets,
+  given the atmospheric conditions and
+  the initial aerosol size distribution and properties.
+See the AerosolActivation module
+  [docs](https://clima.github.io/CloudMicrophysics.jl/dev/Aerosol_activation/)
+  for derivation and description of all input parameters.
```
Comment (odunbar) [inline docs/src/examples/Aerosol_activation_example.md:14]: I think it is ok if you have docs for the aerosol activation code elsewhere, 

However could you write out the parameters and the data, more explicitly? Consider creating a "setting up the inverse problem" section to mimic that of `Cloudy` and `Lorenz`.

**Comment by jakebolewski** [comment]
@odunbar you can, see how we did it in the previous QA Clima repos.  I would say it's a pain  but it should be copy and paste from those examples.

File: docs/src/examples/Aerosol_activation_example.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+```julia
+julia --project
+] instantiate
+```
+
+### Running the Example
+
+The example is written as a
+  [jupyter notebook]((https://github.com/CliMA/EnsembleKalmanProcesses.jl/examples/AerosolActivation/AerosolActivation.ipynb)
```
[the remark this answers is not in the document]
Comment (trontrytel) [reply docs/src/examples/Aerosol_activation_example.md:48]: Now that I think about this, would it even show up before merging? As it is pointing to an address in the main repo that is not there yet?

File: docs/src/examples/Aerosol_activation_example.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  [EnsembleKalmanProcess.jl](https://github.com/CliMA/EnsembleKalmanProcesses.jl),
+  [CLIMAParameters.jl](https://github.com/CliMA/CLIMAParameters.jl) and
+  [CloudMicrophysics.jl](https://github.com/CliMA/CloudMicrophysics.jl).
+To ensure that all the dependencies are met
+  start Julia using the Project.toml file provided in the example.
+Then run the Julia package manager to download all the dependecies.
+
+```julia
+julia --project
+] instantiate
```
[the remark this answers is not in the document]
Comment (trontrytel) [reply docs/src/examples/Aerosol_activation_example.md:42]: This sounds reasonable. Do you want to do it as a separate PR that will be merged before this? Or do you want me to include it in this PR?

File: docs/src/examples/Aerosol_activation_example.md (hunk 2 above)
Comment (trontrytel) [reply docs/src/examples/Aerosol_activation_example.md:14]: will do

**Comment by trontrytel** [comment]
> @odunbar you can, see how we did it in the previous QA Clima repos. I would say it's a pain but it should be copy and paste from those examples.

@odunbar - I think it's not worth the hassle for what we currently have there. I'll convert it to a literate example that will generate some plots to show in the docs and test the convergence to the known true values. I think this will be more useful

File: docs/src/examples/Aerosol_activation_example.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (trontrytel) [reply docs/src/examples/Aerosol_activation_example.md:42]: You have sections about how to run examples in other examples in the docs: https://clima.github.io/EnsembleKalmanProcesses.jl/dev/examples/lorenz_example/#Running-the-Example, https://clima.github.io/EnsembleKalmanProcesses.jl/dev/examples/Cloudy_example/#Running-the-Example

So all of them should be removed?

**Comment by trontrytel** [comment]
This is a new version of the aerosol activation example. I wrote it hoping I can use the Literate on it. However the example has some additional dependencies and I don't know how to make Literate aware of them. @jakebolewski - is there a way to do it without adding those dependencies to the docs dependencies, but keep them in the examples folder as they are now?

**Comment by trontrytel** [comment]
@odunbar - here is a second try with the docs. This time hoping to use the literate way. 

Could you take a look at the new file? One question I have is that I noticed that the converged results from EKP.jl are slightly different each time I run this. I thought that by specifying the random seed I would always get the same behaviour. Could you double check?

**Comment by trontrytel** [comment]
> @trontrytel for me the docs fail? but i can look at the example file

yep - they will until I figure out how to pass in the dependencies to literate.

**Comment by navidcy** [comment]
Is this PR superseded by #54?

**Pull request closed without merging**